src/newViews/MyProfile/index.tsx

```diff
@@ -1,11 +1,266 @@
+import {
+    ObjectSchema,
+    requiredStringCondition,
+    useForm,
+    createSubmitHandler,
+    arrayCondition,
+} from '@togglecorp/toggle-form';
 
 import styles from './styles.scss';
 
-function MyProfile() {
+const mapStateToProps = (state: AppState) => ({
+    activeUser: activeUserSelector(state),
+});
+
+type EmailOptOut = 'news_and_updates' | 'join_requests' | 'email_comment';
+
+interface User {
+    id: number;
+    username: string;
+    firstName: string;
+    lastName: string;
+    displayName: string;
+    lastActiveProject: number;
+    loginAttempts: number;
+    email: string;
+    organization: string;
+    displayPicture: number;
+    displayPictureUrl: string;
+    language: string;
+    emailOptOuts: EmailOptOut[];
+}
+
+type FormType = Partial<Pick<User, 'firstName' | 'lastName' | 'organization' | 'language' | 'emailOptOuts'>>;
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        firstName: [requiredStringCondition],
+        lastName: [requiredStringCondition],
+        organization: [],
+        language: [requiredStringCondition],
+        emailOptOuts: [arrayCondition],
+    }),
+};
+
+type EmailOptOutOption = {
+    key: EmailOptOut;
+    label: string;
+}
+const emailOptOutsOptions: EmailOptOutOption[] = [
+    { key: 'news_and_updates', label: _ts('userProfile', 'newsAndUpdatesInfo') },
+    { key: 'join_requests', label: _ts('userProfile', 'joinRequestsInfo') },
+    { key: 'email_comment', label: _ts('userProfile', 'entryCommentsInfo') },
+];
+const emailOptOutKeySelector = (d: EmailOptOutOption) => d.key;
+const langaugeKeySelector = (d: LanguagePreference) => d.code;
+const languageLabelSelector = (d: LanguagePreference) => d.title;
+
+const initialValue: FormType = {};
+
+
+interface Props {
+    activeUser: { userId: number };
+}
+
+function MyProfile(props: Props) {
+    const {
+        activeUser,
+    } = props;
+
+    const {
+        pristine,
+        value,
+        error,
+        onValueChange,
+        validate,
+        onErrorSet,
+        onValueSet,
+    } = useForm(initialValue, schema);
+
+    const {
+        pending: userGetPending,
+    } = useRequest<User>({
+        url: `server://users/${activeUser.userId}/`,
+        method: 'GET',
+        onSuccess: (response) => {
+            onValueSet(response ?? {});
+            onErrorSet({});
+        },
+        onFailure: (_, errorBody) => {
+            notifyOnFailure(_ts('myProfile', 'myProfileTitle'))({ error: errorBody });
+        },
+    });
+    const {
+        pending: languagesPending,
+        response: languageResponse,
+    } = useRequest<MultiResponse<LanguagePreference>>({
+        url: 'server://languages/',
+        method: 'GET',
+        onFailure: (_, errorBody) => {
+            notifyOnFailure(_ts('myProfile', 'myProfileTitle'))({ error: errorBody });
+        },
```

Comment:
We have a better way to do this. Maybe wait for https://github.com/the-deep/client/pull/1632 ? @AdityaKhatri @samshara